src/emulator/controller.ts

```diff
@@ -75,8 +75,22 @@ export async function cleanShutdown(): Promise<boolean> {
   return true;
 }
 
+export function filterEmulatorTargets(options: any): string[] {
+  let targets = VALID_EMULATOR_STRINGS.filter((e) => {
+    return options.config.has(e) || options.config.has(`emulators.${e}`);
+  });
+
+  if (options.only) {
+    targets = _.intersection(targets, options.only.split(","));
+  } else if (options.except) {
+    targets = _.difference(targets, options.except.split(","));
+  }
```

Comment:
should we throw an error if someone tries `--only database --except firestore`? right now it seems like we just ignore the `--except` if you provide both

Reply:
Fun fact: `emulators:start` and `exec` don't even accept `--except` for now so I will just drop this branch.